packages/next/build/check-next-script-import.ts

```diff
@@ -0,0 +1,16 @@
+import { promises } from 'fs'
+import { join } from 'path'
+
+export async function checkNextScriptImport(
+  pagePaths: string[],
+  pagesDir: string
+): Promise<boolean> {
+  for (const page of pagePaths) {
+    const fileContent = await promises.readFile(join(pagesDir, page), 'utf8')
+    if (/import[^}]*.*('|")next\/script('|")/.test(fileContent)) {
+      return true
+    }
+  }
```

Comment:
Is there a better way to detect `next/script` usage during build? I first tried writing a webpack plugin that checks for the script module after all modules have finished compiling, but quickly realized I can't use `DefinePlugin` to define env variables _after_ compilation only before.

Reply:
Maybe we could implement a build analysis just like: `next/build/analysis/extract-const-value`, `next/build/analysis/get-page-static-info` and `next/build/analysis/parse-module`.

Reply:
Can we utilize SWC? 

https://github.com/vercel/next.js/pull/37589

Reply:
Ooh this is great. I'll hold off on this PR until that gets merged and use `scanNextPackageUsages` instead